### PR TITLE
Multichannel Marketing Integration

### PR DESCRIPTION
**Pull request by nima-karimi**
### Changes proposed in this Pull Request:

Related to https://github.com/woocommerce/woocommerce/issues/34548.

This PR implements an integration with the Multichannel Marketing library that we're building in WooCommerce core. The MCM library is still in development (https://github.com/woocommerce/woocommerce/pull/35099) and is not merged into WC core yet. I've chosen GLA as the first test subject since I had the most experience with and the MCM design (pe2C5g-1g-p2) was made with GLA in mind.

### Detailed test instructions:

Note: To test this PR, you need to checkout and build the [feature/34549-multichannel-marketing-framework](https://github.com/woocommerce/woocommerce/tree/feature/34549-multichannel-marketing-framework) branch of WC core. (This is why the automated unit tests are failing in GitHub)

1. Go to `WooCommerce Settings > Advanced > Features` and turn on the multichannel marketing feature.
2. Go to the Marketing > Overview page and confirm Google Listings & Ads appears in the "Installed marketing extensions" section
3. Ensure you have **NOT** set up the extension or completed onboarding. If you have, reset the extension data.
4. Confirm that a link to "Finish setup" is displayed next to each extension name if it has not been set up and links to their onboarding/setup process.
5. Once you complete setup/onboarding, confirm that the "Finish setup" link is replaced with the "Settings" link and points to the extension's settings page.

Extra test to confirm that the integration will be disabled if WooCommerce is on a previous version:

- Checkout and build the `trunk` branch of WooCommerce core and confirm that the `GLAChannel` class is not registered. You can check this by either setting a xDebug break point on the class construct, or simply changing the value returned by `get_name` to something else and confirming that the name displayed in "Installed Extensions" is different.

### Additional details:
- The GLAChannel class will only be registered if the `WC_MCM_EXISTS` constant is previously defined by the WooCommerce core. This is to ensure there are no errors with previous versions of WC that don't include the required interface/classes.

### Changelog entry

> Add - Integration with WooCommerce Multichannel Marketing

### CODE REVIEW

File: src/MultichannelMarketing/MarketingChannelRegistrar.php
Comment (layoutd) [inline src/MultichannelMarketing/MarketingChannelRegistrar.php]: Could `wc_get_container` be moved to the `WC` proxy?

**Review by layoutd — APPROVED** [review]
This looks good, also ran the tests locally against the feature branch of WC. I left one small question regarding global functions vs proxy, and have this clarification of the testing instructions:

❔ This shows "Manage" instead of "Settings":
> 5. Once you complete setup/onboarding, confirm that the "Finish setup" link is replaced with the "Settings" link and points to the extension's settings page.

File: src/MultichannelMarketing/MarketingChannelRegistrar.php
Comment (nima-karimi) [reply src/MultichannelMarketing/MarketingChannelRegistrar.php]: Yes, good idea. Moved it in https://github.com/woocommerce/google-listings-and-ads/pull/1730/commits/4e33270cadbf1d150fd1fd36a78c32ccb3021229

**Comment by nima-karimi** [comment]
Thanks for the review, @layoutd.

I won't merge this PR because the [Woo core MCM feature PR](https://github.com/woocommerce/woocommerce/pull/35099) has not yet been merged and released. It's best only to merge this when the core feature is merged in Woo.

> This shows "Manage" instead of "Settings":

Yes, "Manage" is the correct wording.

**Review by layoutd — CHANGES_REQUESTED** [review]
Just circling back to make a note that before merging, this PR should be updated to reflect the changes in https://github.com/woocommerce/woocommerce/pull/36012 (specifically, renaming `get_error_no` to `get_error_count`)

**Review by layoutd — APPROVED** [review]
👍🏻  The last round of changes looks good! Got the tests running with 
```bash
rm -rf $TMPDIR/wordpress*
bin/install-wp-tests.sh DBNAME DBUSER DBPW localhost latest feature/34556-marketing-api
vendor/bin/phpunit
```

File: src/MultichannelMarketing/GLAChannel.php
Comment (layoutd) [inline src/MultichannelMarketing/GLAChannel.php:202]: ⚠️ I believe this should be `dashboard` instead of `settings` (after testing for woocommerce/woocommerce#36222)

File: src/MultichannelMarketing/GLAChannel.php
Comment (layoutd) [inline src/MultichannelMarketing/GLAChannel.php]: ⚠️ This URL doesn't seem right (after testing for woocommerce/woocommerce#36222):

```suggestion
						admin_url( 'admin.php?page=wc-admin&path=/google/dashboard&subpath=/campaigns/edit&programId=' . $campaign_data['id' ),
```

File: src/MultichannelMarketing/GLAChannel.php
Comment (layoutd) [inline src/MultichannelMarketing/GLAChannel.php]: ⚠️ I believe this should be `dashboard` instead of `settings` (after testing for woocommerce/woocommerce#36222)

**Review by layoutd — CHANGES_REQUESTED** [review]
Sorry, while testing the WooCommerce MCM API PR, I came across a couple of (what I understand are) incorrect URLs.

**Comment by nima-karimi** [comment]
Ooh, nice catch. Fixed those URLs in https://github.com/woocommerce/google-listings-and-ads/pull/1730/commits/8eafec351369c5ca499c0ea6340898fa829d63fc

File: src/MultichannelMarketing/GLAChannel.php
Comment (layoutd) [inline src/MultichannelMarketing/GLAChannel.php]: I think this one is still missing one parameter: `&programId=19557946999`

So it should be, I think,
```php
admin_url( 'admin.php?page=wc-admin&path=/google/settings&subpath=/campaigns/edit&programId=' . $campaign_data['id'] )
```

**Review by layoutd — CHANGES_REQUESTED** [review]
One last change 🥺

File: src/MultichannelMarketing/GLAChannel.php
Comment (nima-karimi) [reply src/MultichannelMarketing/GLAChannel.php]: Makes sense 😅 Added in https://github.com/woocommerce/google-listings-and-ads/pull/1730/commits/331c4856df35cef56806e61b210e1704e7170db5

**Review by layoutd — APPROVED** [review]
👍🏻  LGTM, thanks @nima-karimi!

For reference, I got the tests working locally with
```bash
rm -rf $TMPDIR/wordpress*
bin/install-wp-tests.sh TESTDB DBUSER DBPASS localhost latest feature/34548-multichannel-marketing-backend
vendor/bin/phpunit
```

**Comment by nima-karimi** [comment]
The Multichannel feature backend and its APIs were [released with WooCommerce 7.4 on Feb. 14, 2023](https://developer.woocommerce.com/2023/02/14/woocommerce-7-4-released/). Merging this PR now.

**Pull request merged**
